wagtail/wagtailforms/models.py

```diff
@@ -132,15 +132,37 @@ class AbstractForm(Page):
 
     form_builder = FormBuilder
 
-    def __init__(self, *args, **kwargs):
-        super(AbstractForm, self).__init__(*args, **kwargs)
-        if not hasattr(self, 'landing_page_template'):
-            template_wo_ext = re.match(HTML_EXTENSION_RE, self.template).group(1)
-            self.landing_page_template = template_wo_ext + '_landing.html'
+    @property
+    def landing_page_template(self):
+        HTML_EXTENSION_RE = re.compile(r"(.*)\.html")
+        template_wo_ext = re.match(HTML_EXTENSION_RE, self.template).group(1)
+        warnings.warn(
+            "The automatically generated landing_page_template attribute is deprecated - "
+            "use AbstractForm.get_template(request, is_landing_page=True), or "
+            "set an explicit landing_page_template attribute instead.",
+            RemovedInWagtail13Warning)
+        return template_wo_ext + '_landing.html'
 
     class Meta:
         abstract = True
 
+    def get_template(self, request, is_landing_page=False, *args, **kwargs):
```

Comment:
I think it might be a bit cleaner to split this into two functions, ``get_template`` and ``get_landing_page_template``. As it looks like that'll cleanup some of the internal logic a bit.